src/tasks/task-module.js

```diff
@@ -204,7 +206,7 @@ class TaskModule extends BaseModule {
       if (taskInfo.failedCount >= failAfterAttempts)
         return;
 
-      if (!taskKlass.shouldRun(taskIndex, lastTime, currentTime, diff))
+      if (DateTime.fromJSDate(taskInfo.nextRunAt) > DateTime.now())
```

Comment:
I would like `nextRunAt` to support a. a timestamp (native millisecond resolution), b. a `Date` instance, and c. a `DateTime` instance. Please update the code to make the proper checks, and to take the correct path based on the type of `nextRunAt`. We should also have this check be `greater or equal`.

It appears `Date`, `DateTime`, and a number primitive will all behave the same when prefixed with a `+`. You might be able to leverage this. For example:

`if (+taskInfo.nextRunAt >= DateTime.now().toMillis()) ...`